### PR TITLE
Pull request from development to master with all requirements

### PR DESCRIPTION
**Pull request by hechukwu**
#### What does this PR do?

This PR is merges the developed application to the master branch

#### Description of Task to be completed?

The inverted index application completed with all requirements.

#### How can this be manually tested?

By Clicking [here](https://indexinverted.herokuapp.com/)

### CODE REVIEW

File: src/js/app.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+     * @param {Object}  object to check
+     * @return {Int}  true or flase
+     */
+    $scope.length = object => Object.keys(object).length;
+
+
+    /**  configures the alert property
+     *
+     * @param {Strimg} message to alert
+     * @param {tye} type of message
```
Comment (abdulsemiu-atanda) [inline src/js/app.js]: typo here...you mean `type` right?

Comment (hechukwu) [reply src/js/app.js]: Thanks, yeah

File: spec/invertedIndexTest.spec.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        will: [0],
+        you: [0, 1],
+        your: [0, 1] });
+    }));
+
+  describe('Get Index',
+  () => it('should return undefined if no index has been created', () => {
+    const getIndex = newIndex.getIndex('book3');
+    expect(getIndex).toBeUndefined();
+  }));
```
Comment (ghost) [inline spec/invertedIndexTest.spec.js:98]: Write a test case to check that the index of a valid book is actually gotten

File: src/InvertedIndex.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  /** gets an indexed file
+   * takes an indexed file as argument and return the indexes of the
+   * file
+   * @param{String} fileName - The name of the file whose index is required
+   * @return{Object} index - The correct mapping of words
+   *  to locations for specified file
+   */
+  getIndex(fileName) {
+    const file = this.index[fileName];
+    return file || undefined;
```
Comment (ghost) [inline src/InvertedIndex.js:81]: Return a better error message rather than undefined

File: src/InvertedIndex.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+  /** search inverted index
+   * takes a string of words to search with the fileName
+   * and returns an object with words mapped
+   * to document locations
+   * @param{String} searchQuery - Words to search for
+   * @param{String} fileName - file to query
+   * @return{Object} searchResults - Maps searched words to document locations
+   */
+  searchIndex(searchQuery, fileName) {
```
Comment (ghost) [inline src/InvertedIndex.js:92]: Can users search through the index without the filename? `searchIndex(searchQuery)` should be valid. Handle a case where users can enter search query as an array e.g `myindex.searchIndex(['alice', ['in', ['wonder']], 'land', 'myFileName')`.
Write tests for the above as well

File: src/js/app.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+    $scope.uploadedFiles = {};
+
+    $scope.index = [];
+
+    $scope.showTable = false;
+
+    $scope.fileCount = [];
+
+    $scope.allIndexedFiles = [];
```
Comment (ghost) [inline src/js/app.js]: The spaces you have after your $scope variables declarations, are they necessary?

File: spec/invertedIndexTest.spec.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    }),
+
+    it('should return correct length of array of words', () => {
+      expect(invertedIndex.tokenize(sampleSentence).length).toBe(10);
+      expect(invertedIndex.uniqueWords(sampleSentence).length).toBe(9);
+    }));
+
+  describe('Populate Index',
+  () => it('should have an Index created', () => expect(newIndex
+  .index.books).toBeDefined()),
```
Comment (ghost) [inline spec/invertedIndexTest.spec.js:46]: Write a test that checks that the file passed in is a valid json file

File: README.md [hunk 7]
```diff
@@ -11,7 +15,7 @@
 The purpose of an inverted index is to allow user perform fast and full text search
 
 # **How can I get started with the project**
-* NA
+* You can view the application [here](https://indexinverted.herokuapp.com/)
 
 # **Technologies for the app**
```
Comment (ghost) [inline README.md]: Make your READme more explanatory. You can provide background context and sample. Include more information like: how to use locally, run tests, contribute to the project and any other relevant information

**Comment by hechukwu** [comment]
Thanks @andela-msosan

**Pull request closed without merging**